Apps/US/HybridGP_US/app/src/Codeunits/GPCloudMigrationUS.Codeunit.al

```diff
@@ -44,4 +75,296 @@ codeunit 42004 "GP Cloud Migration US"
                 end;
             until Vendor.Next() = 0;
     end;
+
+    internal procedure IsIRSFormsFeatureEnabled(): Boolean
+    var
+#if not CLEAN25
+        FeatureManagementFacade: Codeunit "Feature Management Facade";
+#endif
+        IsEnabled: Boolean;
+    begin
+        IsEnabled := true;
+
+#if not CLEAN25
+        IsEnabled := FeatureManagementFacade.IsEnabled(IRSFormFeatureKeyIdTok);
+#endif
+
+        exit(IsEnabled);
+    end;
+
+    local procedure SetupIRSFormsFeatureIfNeeded()
+    var
+        GPCompanyAdditionalSettings: Record "GP Company Additional Settings";
+        IRSReportingPeriod: Record "IRS Reporting Period";
+        ReportingYear: Integer;
+    begin
+        if not IsIRSFormsFeatureEnabled() then
+            exit;
+
+        GPCompanyAdditionalSettings.GetSingleInstance();
+        ReportingYear := GPCompanyAdditionalSettings.Get1099TaxYear();
+
+        if not IRSReportingPeriod.Get(Format(ReportingYear)) then begin
+            CreateReportingPeriod(ReportingYear);
+            PopulateFormsAndBoxes(Format(ReportingYear));
+        end;
+    end;
+
+    local procedure CreateReportingPeriod(ReportingYear: Integer)
+    var
+        IRSReportingPeriod: Record "IRS Reporting Period";
+    begin
+        IRSReportingPeriod.Validate("No.", Format(ReportingYear));
+        IRSReportingPeriod.Validate("Starting Date", DMY2Date(1, 1, ReportingYear));
+        IRSReportingPeriod.Validate("Ending Date", DMY2Date(31, 12, ReportingYear));
+        IRSReportingPeriod.Validate("Description", Format(ReportingYear));
+        IRSReportingPeriod.Insert(true);
+    end;
+
+
+    // Copied from internal codeunit 10039 "IRS Forms Data"
+    local procedure PopulateFormsAndBoxes(PeriodNo: Code[20])
```

Comment:
Can you move these to a dedicated codeunit? Maybe name it GP IRS Form data if it is listing the GP data. It would improve the readability.

Reply:
Moved to the new Codeunit